[18.0][MIG] shopfloor - round 1

sebalix opened the pull request:
Starting from not installable module #5 .

Depends on:
- https://github.com/OCA/stock-logistics-workflow/pull/1971
- https://github.com/OCA/stock-logistics-workflow/pull/2031

## TODO

Status of the migration:

- [x] core functions (actions, helpers)
- [x] handling of qty done
- [ ] single_pack_transfer (to be discarded in favor of shopfloor_single_product_transfer)
- [ ] cluster_picking
- [ ] checkout
- [ ] delivery
- [ ] location_content_transfer
- [ ] zone_picking

simahawk committed:
shopfloor: migrate to v18

shopfloor: v18 views

shopfloor: v18 move_line.reserved_qty

shopfloor: v18 misc

shopfloor: v18 mig actions_data_detail

shopfloor: v18 mig actions_data

shopfloor: v18 mig actions_search

shopfloor: v18 mig actions_stock

shopfloor: v18 picking_form

shopfloor: v18 use odoo.tests.Form

simahawk committed:
shopfloor: fix search.find w/o barcode

sebalix committed:
shopfloor: v18 move_line.reserved_uom_qty

This field has been replaced by 'quantity' starting from Odoo 17.0.

sebalix committed:
shopfloor: v18 picking.package_level_ids_details -> picking.package_level_ids

sebalix committed:
shopfloor: adapt test_stock_split for pick+pack+ship

Create manually the pick+pack+ship chained moves. It is required since
Odoo 18.0 generates by default chained moves through push rules.

sebalix committed:
shopfloor: v18 mig <stock.move.line>.replace_package()

sebalix committed:
shopfloor: v18 mig picking._put_in_pack() signature

sebalix commented:
/ocabot migration shopfloor

grindtildeath commented on shopfloor/actions/stock.py:
```diff
@@ -0,0 +1,252 @@
+            if move.state in ("partially_available", "assigned"):
+                quantity -= sum(move.move_line_ids.mapped("quantity_product_uom"))
+            elif move.state in ("done"):
+                quantity -= move.product_uom_qty
```
```suggestion
                quantity -= move.quantity
```
I guess the same field can be used for the condition above

grindtildeath committed:
shopfloor: drop use of ``estimated_pack_weight_kg``

This field does not exists anymore.

cf https://github.com/OCA/stock-logistics-tracking/pull/48

grindtildeath committed:
shopfloor: adapt qty fields

Rely on stock move line qty_picked

shopfloor: Fix calculation of returnable qty on done picking

It must use quantity effectively moved and not original demand

shopfloor: Do not split full quantity of a move

shopfloor: Replace quantity_done by quantity_picked

shopfloor: Rename variable quantity_done to quantity_reserved

grindtildeath committed:
shopfloor: Add field stock.move.line.is_shopfloor_created

simahawk commented:
As per internal discussion here's a PR w/ the module not installable to be able to focus here only on migration changes
https://github.com/OCA/stock-logistics-shopfloor/pull/5

simahawk commented:
Rebased after #5 and removed the test dep on https://github.com/OCA/stock-logistics-workflow/pull/1971 that was merged.

simahawk commented:
My proposal to ease devs for other modules too:
1. merge ASAP w/o scenario
2. list scenario as TODO list in the readme
3. work on each scenario one by one
The same applies to `shopfloor_mobile`.

NOTE: we discussed about splitting scenario out to single modules but it will make harder to fwd/bkp changes to/from v16.

simahawk commented on shopfloor/actions/data.py:
```diff
@@ -217,8 +217,9 @@ def move_lines(self, records, **kw):
             "id",
-            "qty_done",
-            "reserved_uom_qty:quantity",
+            "picked",
```
these changes that affect the response schema are not good.
We should rollback and
* keep `qty_done`, if necessary map it to another field or add a computed field
* drop `picked` and `quantity_picked`

CC @grindtildeath

grindtildeath replied:
Let's check my changes there before having it merged here
https://github.com/camptocamp/stock-logistics-shopfloor/pull/3

grindtildeath committed:
shopfloor: Fix actions migration to rely on qty_picked

simahawk commented on shopfloor/actions/data.py:
```diff
@@ -218,7 +218,8 @@ def _move_line_parser(self):
             "id",
             "qty_done",
-            "reserved_uom_qty:quantity",
+            # FIXME
```
fixme what?

simahawk commented on shopfloor/actions/data.py:
```diff
@@ -227,7 +228,8 @@ def _move_line_parser(self):
                 lambda rec, fname: rec.move_id.priority or "",
             ),
-            "progress",
+            # TODO: progress is not available on move lines anymore, what to do for UI?
```
doesn't come from `stock_picking_progress`?

simahawk commented on test-requirements.txt:
```diff
@@ -0,0 +1 @@
+odoo-addon-stock_move_line_change_lot @ git+https://github.com/OCA/stock-logistics-workflow@refs/pull/1971/head#subdirectory=stock_move_line_change_lot
```
this should be updated

simahawk reviewed (commented):
LGTM overall

simahawk committed:
shopfloor: disable test scenario

We'll migrate scenario 1 by 1.

simahawk committed:
shopfloor: update readme for partial mig on v18

simahawk commented on shopfloor/actions/data.py:
```diff
@@ -358,7 +360,7 @@ def _get_operation_progress(self, domain):
         for line in lines:
             operations_done += line.qty_done
-            operations_to_do += line.reserved_uom_qty - line.qty_done
+            operations_to_do += line.quantity - line.qty_done
```
shouldn't be `line.qty_picked`?

simahawk commented:
/ocabot migration shopfloor

simahawk commented:
> There's no issue in this repo with the title 'Migration to version 18.0' and the milestone 18.0, so not possible to add the comment.

you are drunk bro!

simahawk commented:
Last thing to fix
```
2025-07-31 10:50:59,300 256 WARNING odoo odoo.addons.shopfloor.models.stock_move_line: `_split_pickings_from_source_location` is deprecated and replaced by `_extract_in_split_order` 
```

simahawk committed:
shopfloor: drop stock.move.line_split_pickings_from_source_location

simahawk commented:
> My proposal to ease devs for other modules too:
> 
>     1. merge ASAP w/o scenario
> 
>     2. list scenario as TODO list in the readme
> 
>     3. work on each scenario one by one
>        The same applies to `shopfloor_mobile`.
> 
> 
> NOTE: we discussed about splitting scenario out to single modules but it will make harder to fwd/bkp changes to/from v16.

FTR I reported this strategy on the README of the module and on #13

grindtildeath commented on shopfloor/__manifest__.py:
```diff
@@ -6,7 +6,7 @@
     "name": "Shopfloor",
     "summary": "manage warehouse operations with barcode scanners",
-    "version": "16.0.2.8.0",
+    "version": "18.0.1.0.0",
```
Wondering if we shouldn't have a 18.0.0.x.y as we didn't _fully_ migrate yet :question:

simahawk replied:
Kind of make sense :thinking: :+1:

simahawk replied:
done 0.1.0

grindtildeath reviewed (approved):
I only checked the last few commits and IMO it looks good for a first version, we can still have more fixes coming as we are migrating scenarios and other modules

simahawk commented:
/ocabot merge nobump
